text/0486-deprecate-mouseenter.md

```diff
@@ -0,0 +1,144 @@
+not bubble. In the past it still worked nevertheless as jQuery transparently handled this
+for us, as it had special support for event delegation for these events, essentially by using 
+(bubbling) `mouseover` events to replicate the semantics of `mouseenter`/`mouseleave` events.
+
+When support for jQuery-less apps was introduced, this [left a hole](https://github.com/emberjs/ember.js/issues/16591)
+in the jQuery-less EventDispatcher implementation. But as support for those events was and
+still is part of Ember's pubic API, we had no chance other than to [implement support
+for jQuery-less apps](https://github.com/emberjs/ember.js/pull/16603) using the same 
+`mouseover` based approach.
+
+This however comes with a cost: besides an [unresolved issue](https://github.com/emberjs/ember.js/issues/17228)
+the implementation has some performance drawbacks, as it has to process every `mouseover` event on 
+*any* element, create fake `mouseenter`/`mouseleave` events and try to dispatch them, even when 
+not a single component/action needs them.  
+
+While efforts to make this more "pay as you go" are [possible](https://github.com/emberjs/ember.js/pull/17911), 
+the trade-off of keeping support around still seems unfavorable, as these events fire so
+frequently, while they are (most certainly) very rarely used.
+
+This is even more so given that Glimmer Components with their outerHTML semantics do not 
+work with event handler methods, and `{{action}}` will eventually fade away in favor of
+`{{on}}` using native `addListener()`.
+
+## Transition Path
+
+When a `mouseenter`/`mouseleave` event is dispatched by the EventDispatcher to a component
+or `{{action}}` listening to that event type, a deprecation warning will be issued.
+
+The linked migration guide will cover these use cases:
+
+#### Ember.Component
+
+Before:
+
+```js
+import Component from '@ember/component';
+
+export default Component.extend({
+  mouseEnter(e) {
+    // do something
+  }
+});
+```
+
+After:
+
+```js
+import Component from '@ember/component';
+import { bind } from '@ember/runloop';
+
+export default Component.extend({
+  mouseenter(e) {
+    // do something
+  },
+  didInsertElement() {
+    this._super(...arguments);
+    this._mouseenter = bind(this, this.mouseenter);
+    this.element.addEventListener('mouseenter', handler);
+  },
+  willDestroyElement() {
+    this._super(...arguments);
+    this.element.removeEventListener('mouseenter', this._mouseenter);
+  }
+});
+```
+
+#### `{{action}}` modifier
+
+Before:
+
+```hbs
+<button {{action "handleMouseEnter" on="mouseenter"}}>Hover</button>
+```
+
+After:
+
+```hbs
+<button onmouseenter={{action "handleMouseEnter"}}>Hover</button>
+```
+
+or (based on [RFC471](https://github.com/emberjs/rfcs/blob/master/text/0471-on-modifier.md))
+
+```hbs
+<button {{on "mouseenter" this.handleMouseEnter}}>Hover</button>
+```
+
+## How We Teach This
+
+The deprecation guide should explain the transition path as shown above.
```

Comment:
Could you write out the explanation instead? The goal for this section is that a contributor can write the deprecation guide and update existing documentation by reading this section alone.

Reply:
This would mean just duplicating the before/after examples. Does that make sense really?

Fwiw, I looked at this deprecation RFC when writing this, which does a similar thing: https://github.com/emberjs/rfcs/blob/master/text/0449-deprecate-partials.md

Reply:
Ya, I don't really think it makes sense to duplicate the same content here. If the content needs to be moved into this section instead of transition path I guess thats fine, but it means that the transition path would just say "read the deprecation guide in the "how we teach this" below" which is a bit odd.